test/Npgsql.Tests/DistributedTransactionTests.cs

```diff
@@ -221,12 +231,14 @@ public void Connection_reuse_race_after_transaction([Values(false, true)] bool d
         using var adminConn = OpenConnection();
         var table = CreateTempTable(adminConn, "name TEXT");
 
+        using var dataSource = CreateDataSource(ConnectionStringEnlistOff);
```

Comment:
Yeah, pretty sure you're right, I just went for the more 'stable' approach to make sure everything works correctly.

Reply:
I think it's important to keep some reuse, since that tests that we don't have bugs when using the same data source concurrently. In other words, if every test had its own isolated data source, we'd have prefectly isolated tests but no test coverage for concurrency bugs.

Up to you for this specific case...